src/api/utils.rs

```diff
@@ -1 +1,99 @@
-pub mod utils;
```

Comment:
What rationale have you used for when to use `mod.rs` and when to use `<module-name>.rs`? This one has everything in `utils.rs` and e.g. `execution.rs` was moved to `execution/mod.rs`.

Is it just a matter of size/number of supporting files? E.g. the starting point would be throw everything in `<module-name>.rs`, then when you add an additional file break it out into a `mod.rs` and supporting file? Is it "better" to have module declarations on their own (e.g. `utils/mod.rs` + `utils/supporting-file.rs`)? 

As mentioned elsewhere I have only just read about this, so trying to wrap my head around what the common conventions are.

Reply:
Essentially the way I do it is I use `mod.rs` for modules with multiple files. I.e.

```
src/
  lib.rs
  models/
    mod.rs
    user.rs
```

For single files (like `utils.rs`) I keep them as single files to avoid typing `utils::utils`, etc. If it ever needs to be expanded into multiple files, then I would do like:

```
utils/
  mod.rs
  misc.rs
  reddit.rs
```

[Here is an example project I worked on](https://github.com/andrewzah/danmuji) for a sense of structure. Hope this helps.